ihmc-avatar-interfaces/src/main/java/us/ihmc/avatar/logProcessor/leRobot/LeRobotDatasetDataVariables.java

```diff
@@ -90,45 +98,54 @@ public void addFrame(double timestamp, LeRobotDatasetEpisodeStatistics statistic
 
       for (RobotSide side : handPosesCurrent.sides())
       {
-         pelvisFrame.update(pelvisPoseCurrent::get);
-         framePose.setIncludingFrame(ReferenceFrame.getWorldFrame(), handPosesCurrent.get(side));
-         framePose.changeFrame(pelvisFrame.getReferenceFrame());
-         state.add((float) framePose.getX());
-         state.add((float) framePose.getY());
-         state.add((float) framePose.getZ());
-         state.add((float) framePose.getOrientation().getX());
-         state.add((float) framePose.getOrientation().getY());
-         state.add((float) framePose.getOrientation().getZ());
-         state.add((float) framePose.getOrientation().getS());
-         framePose.setIncludingFrame(ReferenceFrame.getWorldFrame(), forearmPosesCurrent.get(side));
-         framePose.changeFrame(pelvisFrame.getReferenceFrame());
-         state.add((float) framePose.getX());
-         state.add((float) framePose.getY());
-         state.add((float) framePose.getZ());
-         state.add((float) framePose.getOrientation().getX());
-         state.add((float) framePose.getOrientation().getY());
-         state.add((float) framePose.getOrientation().getZ());
-         state.add((float) framePose.getOrientation().getS());
+         if (side == RobotSide.RIGHT || !LeRobotDataset.XYZ_RIGHT_ONLY)
+         {
+            torsoFrame.update(torsoPoseCurrent::get);
+            framePose.setIncludingFrame(ReferenceFrame.getWorldFrame(), handPosesCurrent.get(side));
+            framePose.changeFrame(cameraFrame);
```

Comment:
The state & actions are now in camera frame instead of pelvis frame, which was recommended to me by Jerry.